.lintstagedrc

```diff
@@ -0,0 +1,8 @@
+{
+	"*.ts": [
```

Comment:
Yes there's overlap because all the autoformatting is handled by the first check and only test specific changes by the test specific matching, ie all .ts files should lint correctly and then we run an extra check specifically for tests.  We could split it up so we duplicate the standard linting checks/options in both places but it would make this more complex/less understandable imo and more prone to copy/paste errors where only one is update, for the sake of saving a small amount of time on a commit.

For the time saved I checked and with resin-lint 3.2.0 it would be ~1s per file saved which is actually fairly significant so I optimized resin-lint somewhat and with 3.2.2 it'll be down to ~.5s saved (and I'll probably find some more savings before this)